e2e/playwright/pages/CreatePluginPage.ts

```diff
@@ -0,0 +1,65 @@
+import { type Locator, type Page, expect } from 'playwright/test';
+
+
+
+export class CreatePluginPage{
+
+  readonly page: Page;
+  readonly legalHoldPlugin: Locator;
+  readonly createNewButton: Locator;
+  readonly legalHoldName: Locator;
+  readonly username: Locator;
+  readonly usernameDropdown: Locator;
+  readonly startDate: Locator;
+  readonly legalHoldButton : Locator;
+  readonly modalnotVisible : Locator;
+  
+
+  constructor(page: Page){
+    this.page = page;
+    this.legalHoldPlugin = page.getByRole('link', { name: 'Legal Hold Plugin' });
+    this.createNewButton = page.getByText('create new').first();
+    this.legalHoldName = page.getByPlaceholder('New Legal Hold...');
+    this.username = page.locator('.css-19bb58m')
+    this.usernameDropdown = page.locator('#react-select-2-input');
+    this.startDate = page.getByPlaceholder('Starting from');
+    this.legalHoldButton = page.getByRole('button', { name: 'Create legal hold' })
+    this.modalnotVisible = page.getByText ('Create a new legal hold')
+  }
+
+  async clickLegalHoldPlugin(){
+    await this.legalHoldPlugin.click();
+  }
+
+  async clickCreateNewButton(){
+    await this.createNewButton.click(); 
+    await expect(this.page.getByText('Create a new legal hold')).toBeVisible;
+
+  }
```

Comment:
Same here. Just write the click "action" directly in the spec file (no need to wrap). Then validate (`expect`) the modal to be visible directly in the spec file.